Realm/ObjectStore/object_schema.hpp

```diff
@@ -0,0 +1,50 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2014 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OBJECT_SCHEMA_HPP
+#define REALM_OBJECT_SCHEMA_HPP
+
+#include <string>
+#include <vector>
+
+#include "property.hpp"
+
+namespace realm {
+    class Group;
+    class Table;
+
+    class ObjectSchema {
+    public:
+        ObjectSchema() {}
+
+        // create object schema from existing table
+        // if no table is provided it is looked up in the group
+        ObjectSchema(Group *group, const std::string &name, Table *table = nullptr);
```

Comment:
Wouldn't it be better to split this into two as it covers two use cases: 
1) with string, but no table and 
2) with table but then you don't need the name as you can look that up in the group?